src/lib/y2firewall/clients/auto.rb

```diff
@@ -79,16 +81,18 @@ def summary
       #
       # @param profile [Hash] firewall profile section to be imported
       # @return [Boolean]
-      def import(profile, merge = !Yast::Mode.config)
+      def import(profile, merge = !Yast::Mode.config || !Yast::Mode.auto)
```

Comment:
In autoinstallation the merge was also used as by default each zone contains some predefined services / ports open.

Reply:
But since the import is called by AY at the beginning of the first stage now, it is called before the configuration exists (firewalld is not present at this point). It is one thing I wanted to open a discussion about here ;-)

I also have a version where even import is called from the finish client and there it works just fine.

So as I currently see following options
1) keep it as it is here, but there can be issues with non merging configurations
2) go back to second approach mentioned above, but it was using ```Profile.current```
3) call firewall's import from AY but from different place than others - somewhere around calling finish clients
4) keep things as they are now - keep second stage for firewall mandatory in all cases
5) I missed something

Reply:
> But since the import is called by AY at the beginning of the first stage now, it is called before the configuration exists (firewalld is not present at this point). It is one thing I wanted to open a discussion about here ;-)

If I'm not wrong that was handled by the client. If the package was not installed then it was not imported and tried to import later and that is why the profile was stored as a class variable.

> 
> I also have a version where even import is called from the finish client and there it works just fine.

Yep, that should be the behavior.

> 
> So as I currently see following options
> 
> 1. keep it as it is here, but there can be issues with non merging configurations

Should be merged during the finish write.

> 2. go back to second approach mentioned above, but it was using `Profile.current`
> 3. call firewall's import from AY but from different place than others - somewhere around calling finish clients
> 4. keep things as they are now - keep second stage for firewall mandatory in all cases
> 5. I missed something

Reply:
> If I'm not wrong that was handled by the client. If the package was not installed then it was not imported and tried to import later and that is why the profile was stored as a class variable.

then there was a bug. In short in profile where I wanted firewall to be enabled after installation, the import ended with false (because firewalld was not installed) and later write didn't change anything. I know there is ```import_if_needed``` method, however the firewall simply was not enabled after installation. So, I'm taking note to investigate that.
> 
> > I also have a version where even import is called from the finish client and there it works just fine.
> 
> Yep, that should be the behavior.

i know but worked only if I placed the import into the client explicitly. See above
> 
> > So as I currently see following options
> > 
> > 1. keep it as it is here, but there can be issues with non merging configurations
> 
> Should be merged during the finish write.

ok, as described above, I'll check what is wrong that this didn't work.
>